daffodil-tdml-lib/src/main/scala/org/apache/daffodil/tdml/TDMLAppender.scala

```diff
@@ -38,25 +38,32 @@ class TDMLAppender
     import TDMLAppender._
 
     override def append(event: LogEvent): Unit = {
-        diagnostics = new LogDiagnostic(event.getMessage.getFormattedMessage) +: diagnostics
+        checkForKey
+        diagnostics = diagnostics.updated(Thread.currentThread.getId, diagnostics(Thread.currentThread.getId) :+ new LogDiagnostic(event.getMessage.getFormattedMessage))

```

Comment:
I don't think this line is thread safe? If `append` is called by two different threads at the same time, is it possible that one of the updates will be lost? Can this function be synchronized like the other functions?

Reply:
its done by log4j so it is thread safe but I will add synchronized anyway